src/AstClause.h

```diff
@@ -177,17 +177,7 @@ class AstClause : public AstNode {
 
     /** Add a Literal to the body of the clause */
     void addToBody(std::unique_ptr<AstLiteral> l) {
-        if (dynamic_cast<AstAtom*>(l.get()) != nullptr) {
-            atoms.emplace_back(static_cast<AstAtom*>(l.release()));
-        } else if (dynamic_cast<AstNegation*>(l.get()) != nullptr) {
-            negations.emplace_back(static_cast<AstNegation*>(l.release()));
-        } else if (dynamic_cast<AstProvenanceNegation*>(l.get()) != nullptr) {
-            provNegations.emplace_back(static_cast<AstProvenanceNegation*>(l.release()));
-        } else if (dynamic_cast<AstConstraint*>(l.get()) != nullptr) {
-            constraints.emplace_back(static_cast<AstConstraint*>(l.release()));
-        } else {
-            assert(false && "Unsupported literal type!");
-        }
+        bodyLiterals.emplace_back(l.release());
```

Comment:
Rather than releasing ownership, you can move unique pointer
```suggestion
        bodyLiterals.emplace_back(std::move(literal));
```